tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -367,6 +367,7 @@ def test_raster_metadata_property(self):
 
     def test_raster_info_accessor(self):
         infos = self.rs.info
+        nodata_value = 15 if GDAL_VERSION < (3, 4, 1) else 10
```

Comment:
I checked the tests results with GDAL < 3.4, 3.4.0, 3.4.1, 3.4.2. This can be related with: 
- https://github.com/matthew-baran/gdal/commit/969092a98ed0405791273b8d108b2d6b1ad50aa9, or
- https://github.com/matthew-baran/gdal/commit/b547152546b5cfddbdd0f1163ebb59a82b0733eb

Reply:
this is probably the same as https://github.com/OSGeo/gdal/issues/5431 , that is GDAL >= 3.4.1 will use the nodata value from the .aux.xml side car file when it is present.
another way would be to make sure you don't have a .aux.xml file

Reply:
gdalinfo of GDAL 3.4.2 reports "NoData Value=15" as expected when downloading that file, independently of your test suite. Are you really sure nothing in your test suite creates a .aux.xml side car file (e.g by setting the nodata value to 10 when opening the file in read-only mode) ?

Reply:
@claudep I noticed that `test_read_mode_error()` leaves `gis_tests/data/rasters/raster.tif.aux.xml`, with `nodata_value=10`.